File: Arrays&Hashing/hasDuplicate.cpp
#include <iostream>
#include <vector>
#include <unordered_set>
using namespace std;

// Your original class
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        unordered_set<int> set;
        for (int num : nums) {
            if (set.count(num)) {
                return true;
            } else {
                set.insert(num);
            }
        }
        return false;
    }
};

int main() {
    Solution solution;
    vector<int> nums = {1, 2, 3, 4, 5, 1}; // Change this to test other cases

    bool result = solution.hasDuplicate(nums);
    cout << (result ? "true" : "false") << endl;

    return 0;
}

File: Arrays&Hashing/hasDuplicate.py
# Given an integer array nums, return true if any value appears 
# more than once in the array, otherwise return false.

class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        myset = set()
        for num in nums:
            if num in myset:
                return True
            myset.add(num)
        return False